src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1003,27 +1003,32 @@ public boolean isFeedbackSessionHasQuestionForStudents(
         return !allQuestions.isEmpty();
     }
 
+    public boolean isFeedbackSessionCompletedByStudent(FeedbackSessionAttributes fsa,
+                                                       String userEmail)
+                   throws EntityDoesNotExistException {
+        Assumption.assertNotNull(fsa);
+        if (fsa.respondingStudentList.contains(userEmail)) {
+            return true;
+        }
+        
+        String feedbackSessionName = fsa.feedbackSessionName;
+        String courseId = fsa.courseId;
+        List<FeedbackQuestionAttributes> allQuestions =
+                fqLogic.getFeedbackQuestionsForStudents(feedbackSessionName, courseId);
+        // if there is no question for students, session is complete
+        return allQuestions.isEmpty();
+    }
+
     public boolean isFeedbackSessionCompletedByStudent(
```

Comment:
It's used in test (testing for access to non-existent feedback session) but not anywhere else. I think it's safe to remove?